Content.Server/Psionics/Abilities/MetapsionicPowerSystem.cs

```diff
@@ -0,0 +1,175 @@
+
+        private void OnWidePowerUsed(EntityUid uid, MetapsionicPowerComponent component, WideMetapsionicPowerActionEvent args)
+        {
+            if (HasComp<PsionicInsulationComponent>(uid))
+                return;
+
+            if (!TryComp<PsionicComponent>(uid, out var psionic))
+                return;
+
+            foreach (var entity in _lookup.GetEntitiesInRange(uid, component.Range))
+            {
+                if (HasComp<PsionicComponent>(entity) && entity != uid && !HasComp<PsionicInsulationComponent>(entity) &&
+                    !(HasComp<ClothingGrantPsionicPowerComponent>(entity) && Transform(entity).ParentUid == uid))
+                {
+                    _popups.PopupEntity(Loc.GetString("metapsionic-pulse-success"), uid, uid, PopupType.LargeCaution);
+                    args.Handled = true;
+                    return;
+                }
+            }
+            _popups.PopupEntity(Loc.GetString("metapsionic-pulse-failure"), uid, uid, PopupType.Large);
+            _psionics.LogPowerUsed(uid, "metapsionic pulse", psionic, 2, 4);
+            UpdateActions(uid, component);
+            args.Handled = true;
+        }
+
+        private void OnFocusedPowerUsed(FocusedMetapsionicPowerActionEvent args)
+        {
+            if (!TryComp<PsionicComponent>(args.Performer, out var psionic))
+                return;
+
+            if (HasComp<PsionicInsulationComponent>(args.Target))
+                return;
+
+            if (!TryComp<MetapsionicPowerComponent>(args.Performer, out var component))
+                return;
+
+            var ev = new FocusedMetapsionicDoAfterEvent(_gameTiming.CurTime);
+
+            _doAfterSystem.TryStartDoAfter(new DoAfterArgs(EntityManager, args.Performer, component.UseDelay - psionic.Amplification, ev, args.Performer, args.Target, args.Performer)
+            {
+                BlockDuplicate = true,
+                BreakOnUserMove = true,
+                BreakOnTargetMove = true,
+                BreakOnDamage = true,
+            }, out var doAfterId);
+
+            component.DoAfter = doAfterId;
+
+            _popups.PopupEntity(Loc.GetString("focused-metapsionic-pulse-begin", ("entity", args.Target)), args.Performer, PopupType.Medium);
+
+            _audioSystem.PlayPvs(component.SoundUse, args.Performer, AudioParams.Default.WithVolume(8f).WithMaxDistance(1.5f).WithRolloffFactor(3.5f));
+            _psionics.LogPowerUsed(args.Performer, "focused metapsionic pulse", psionic,
+                (int) MathF.Round(3 * psionic.Amplification - psionic.Dampening),
+                (int) MathF.Round(6 * psionic.Amplification - psionic.Dampening));
+            args.Handled = true;
+
+            UpdateActions(args.Performer, component);
+        }
+
+        private void OnDoAfter(EntityUid uid, MetapsionicPowerComponent component, FocusedMetapsionicDoAfterEvent args)
+        {
+            component.DoAfter = null;
+
+            if (args.Target == null) return;
```

Comment:
You need to check whether the do-after was cancelled, because this event gets fired regardless of do-after result.